### PR TITLE
fix(opencode): inherit user-selected models

### PR DESCRIPTION
**Pull request by dajiaohuang**
## What Changed

- Removed provider-specific `model` and `small_model` pins from the reference OpenCode config.
- Removed per-agent model overrides so primary agents use OpenCode's global model selection and subagents inherit the invoking primary agent's model.
- Added a regression test covering the root config and every registered agent.
- Updated the OpenCode documentation and support matrix to describe the inherited model behavior.

## Why This Change

Context: #2617

The checked-in reference config currently requires Anthropic model IDs even when a user has connected a different OpenCode provider. That makes agents such as `code-reviewer` fail before they can run.

This Draft takes the smallest reversible approach supported by OpenCode's model-selection semantics: omit model overrides and let OpenCode use the user's configured selection.

### Assumptions and boundaries

- Assumes the user has connected a provider and selected a usable model in OpenCode.
- Applies one selected model across ECC agents through OpenCode inheritance. It intentionally gives up the current fixed Opus/Sonnet/Haiku tiering in exchange for provider neutrality.
- Does not inspect credentials, probe provider APIs, add dependencies, or expand the guided installer.
- Does not attempt feature parity with Claude Code.

### Alternative for review

If explicit quality/cost tiering is required, the follow-up alternative is to add provider/model inputs to the installer and generate validated root and per-agent mappings. That is a larger interface and configuration change; this Draft keeps that option open without requiring it for provider-neutral defaults.

OpenCode documents configured model selection and per-agent model inheritance here:
- https://opencode.ai/docs/models
- https://opencode.ai/docs/agents/

## Testing Done

- [ ] Manual testing completed
- [ ] Automated tests pass locally (`node tests/run-all.js`)
- [x] Edge cases considered and tested

Focused validation:
- `node tests/opencode-config.test.js` — 4 passed, 0 failed
- JSON parse validation — passed
- `git diff --check` — passed
- `node tests/run-all.js` — 3716 passed, 20 failed. The failures are confined to Windows environment constraints (missing `/bin/bash`/WSL, unavailable symlink privileges, and unrelated local temp/server assumptions); the OpenCode config, plugin, tools, and build tests pass.

## Type of Change

- [x] `fix:` Bug fix
- [ ] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [x] `docs:` Documentation
- [x] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist

- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [x] Shell scripts pass shellcheck (if applicable)
- [x] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## If you changed dependencies or `package.json` (`bin` / `files` / deps)

Not applicable.

## If you added a skill, command, agent, hook, or CLI tool

Not applicable.

## Documentation

- [x] Updated relevant documentation
- [ ] Added comments for complex logic
- [x] README updated (if needed)

### CODE REVIEW

**Comment by dajiaohuang** [comment]
Regarding the Greptile yarn.lock finding: it does not apply to the current PR head. This PR does not modify yarn.lock; the checked-in lockfile is Yarn 4 metadata v8, package.json pins Yarn 4.9.2, and corepack yarn install --immutable succeeds without rewriting it. No lockfile change is needed.

File: tests/opencode-config.test.js [hunk 1]
```diff
@@ -28,6 +28,19 @@ const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
...
 let failed = 0;
 
+if (
+  test('model selection inherits the user configured OpenCode provider', () => {
+    assert.ok(!Object.hasOwn(config, 'model'), 'Root config must not pin a provider-specific model');
+    assert.ok(!Object.hasOwn(config, 'small_model'), 'Root config must not pin a provider-specific small model');
+
+    for (const [agentId, agent] of Object.entries(config.agent || {})) {
+      assert.ok(!Object.hasOwn(agent, 'model'), `Agent "${agentId}" must inherit the selected OpenCode model`);
+    }
```
[the remark this answers is not in the document]
Comment (dajiaohuang) [reply tests/opencode-config.test.js]: Addressed in `4163411a`. The regression now asserts that `config.agent` is a non-array object with at least one registered agent before iterating directly over `Object.entries(config.agent)`. Re-verified with `node tests/opencode-config.test.js` (4 passed, 0 failed); the current PR CI matrix is green.

**Comment by affaan-m** [comment]
Superseded by #2863 (merged), which consolidated this work with original commits and authors retained. Closing per the consolidation PR.
